graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -268,47 +304,128 @@ def enter(self, node, key, parent, path, ancestors):
                 return None
-            else:  # Name changed, return new node, `visit` will make shallow copies along path
+            else:
+                # Name changed or suppressed, return new node, `visit` will make shallow copies
+                # along path
                 return renamed_node
         else:
             # All Node types should've been taken care of, this line should never be reached
             raise AssertionError('Unreachable code reached. Missed type: "{}"'.format(node_type))
 
 
 class RenameQueryTypeFieldsVisitor(Visitor):
-    def __init__(self, renamings, query_type):
+    def __init__(self, renamings: Dict[str, Optional[str]], query_type: str):
         """Create a visitor for renaming fields of the query type in a schema AST.
 
         Args:
-            renamings: Dict[str, str], from original field name to renamed field name. Any
+            renamings: Dict[str, Optional[str]], from original field name to renamed field name. Any
                        name not in the dict will be unchanged
             query_type: str, name of the query type (e.g. RootSchemaQuery)
         """
         # Note that as field names and type names have been confirmed to match up, any renamed
-        # field already has a corresponding renamed type. If no errors, due to either invalid
-        # names or name conflicts, were raised when renaming type, no errors will occur when
+        # query type field already has a corresponding renamed type. If no errors, due to either
+        # invalid names or name conflicts, were raised when renaming type, no errors will occur when
         # renaming query type fields.
         self.in_query_type = False
         self.renamings = renamings
         self.query_type = query_type
 
-    def enter_object_type_definition(self, node, *args):
+    def enter_object_type_definition(self, node: ObjectTypeDefinitionNode, *args):
         """If the node's name matches the query type, record that we entered the query type."""
         if node.name.value == self.query_type:
             self.in_query_type = True
 
-    def leave_object_type_definition(self, node, key, parent, path, ancestors):
+    def leave_object_type_definition(
+        self, node: ObjectTypeDefinitionNode, key, parent, path, ancestors
+    ):
         """If the node's name matches the query type, record that we left the query type."""
+        if not node.fields:
+            raise CascadingSuppressionError(
+                f"Type renamings {self.renamings} suppressed every type in the schema so it will be"
+                f" impossible to query for anything. To fix this, check why the `renamings` "
+                f"argument of `rename_schema` mapped every type to None. Note that adding "
+                f"suppressions may lead to other types, fields, unions, etc. requiring suppression "
+                f"so you may need to iterate on this before getting a legal schema."
```

Comment:
I'm a bit confused about why this last bit is here describing the workflow for additional suppressions. Don't we want the user to _remove_ suppressions to fix the error? This feels like we're giving the user some mixed messages.

Reply:
Oops, yes-- adding suppressions was true for cascading suppression issues but this is a `SchemaTransformError` so we can remove that.